[Unity] nn.Module Module and Effect

cyx-6 opened the pull request:
This PR introduces the Module and Effect in the nn frontend. It adds the built-in `Linear`, `RMSNorm` and `Embedding` modules. And it adds a `KVCache` effect.

Also, this PR fix the interface of `R.nn.rms_norm` to support optional `bias`.

junrushao reviewed (approved):
LGTM!

junrushao commented:
There's some issue with the CI. I squashed and rebased this PR trying to make this work

junrushao committed:
[Unity] nn.Module Module and Effect

This PR introduces the Module and Effect in the nn frontend. It adds the built-in `Linear`,
`RMSNorm` and `Embedding` modules. And it adds a `KVCache` effect.

Also, this PR fix the interface of `R.nn.rms_norm` to support optional `bias`.
